### PR TITLE
Publish build artifacts to Azure blob storage

### PR DESCRIPTION
**Pull request by ivan-nosar**
This PR adds new step with publishing of build artifacts into BLOB storage. Also additional structural refactoring was performed too.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -53,15 +58,36 @@ jobs:
...
+      inputs:
+        targetType: filePath
+        filePath: "./bin/ci/az-publish.sh"
+      env:
+        AZURE_STORAGE_ACCOUNT: $(AzureStorageAccount)
+        AZURE_STORAGE_KEY: $(AzureStorageKey)
+        AZURE_STORAGE_CONNECTION_STRING: $(AzureStorageConnectionString)
+        SOURCE_BRANCH: $(Build.SourceBranch)
+      displayName: "Publish to Azure Blob Storage"
+      condition: and(succeeded(), eq(variables['Agent.JobName'], variables['AzurePublishWhen']), startsWith(variables['Build.SourceBranch'], 'master'))
```
Comment (jmoody) [inline azure-pipelines.yml]: Add `IndividualCI` to this condition.

File: bin/ci/az-publish.sh [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+az --version
+
+WORKING_DIR="${BUILD_SOURCESDIRECTORY}"
+
+# Upload `CalWebViewApp.app` (zipped)
+APP_ZIP="${WORKING_DIR}/CalWebViewApp/Products/app-cal/CalWebView-cal.app.zip"
+zip_with_ditto "${WORKING_DIR}/CalWebViewApp/Products/app-cal/CalWebView-cal.app" "${APP_ZIP}"
+APP_NAME="CalWebView-${VERSION}-Xcode-${XC_VERSION}-${GIT_SHA}.app.zip"
+azupload "${APP_ZIP}" "${APP_NAME}"
```
[the remark this answers is not in the document]
Comment (ivan-nosar) [reply bin/ci/az-publish.sh]: Already done

File: bin/ci/az-publish.sh [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+# Upload `CalWebViewApp.ipa`
+IPA="${WORKING_DIR}/CalWebViewApp/Products/ipa-cal/CalWebView-cal.ipa"
+IPA_NAME="CalWebView-${VERSION}-Xcode-${XC_VERSION}-${GIT_SHA}.ipa"
+azupload "${IPA}" "${IPA_NAME}"
+
+# Upload `CalWebViewApp.ipa.dSYM` (zipped)
+IPA_DSYM_ZIP="${WORKING_DIR}/CalWebViewApp/Products/ipa-cal/CalWebView-cal.app.dSYM.zip"
+zip_with_ditto "${WORKING_DIR}/CalWebViewApp/Products/ipa-cal/CalWebView-cal.app.dSYM" "${IPA_DSYM_ZIP}"
+IPA_DSYM_NAME="CalWebView-${VERSION}-Xcode-${XC_VERSION}-${GIT_SHA}.ipa.dSYM.zip"
```
[the remark this answers is not in the document]
Comment (ivan-nosar) [reply bin/ci/az-publish.sh]: Already done

File: azure-pipelines.yml (hunk 1 above)
Comment (ivan-nosar) [reply azure-pipelines.yml]: Done

File: bin/ci/az-publish.sh [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+# Upload `CalWebViewApp.app` (zipped)
+APP_ZIP="${WORKING_DIR}/CalWebViewApp/Products/app-cal/CalWebView-cal.app.zip"
+zip_with_ditto "${WORKING_DIR}/CalWebViewApp/Products/app-cal/CalWebView-cal.app" "${APP_ZIP}"
+APP_NAME="CalWebView-${VERSION}-Xcode-${XC_VERSION}-${GIT_SHA}.app.zip"
+azupload "${APP_ZIP}" "${APP_NAME}"
+
+# Upload `CalWebViewApp.app.dSYM` (zipped)
+APP_DSYM_ZIP="${WORKING_DIR}/CalWebViewApp/Products/app-cal/CalWebView-cal.app.dSYM.zip"
+zip_with_ditto "${WORKING_DIR}/CalWebViewApp/Products/app-cal/CalWebView-cal.app.dSYM" "${APP_DSYM_ZIP}"
+APP_DSYM_NAME="CalWebView-${VERSION}-Xcode-${XC_VERSION}-${GIT_SHA}.app.dSYM.zip"
```
[the remark this answers is not in the document]
Comment (ivan-nosar) [reply bin/ci/az-publish.sh]: Done

**Pull request merged**